agent/task.go

```diff
@@ -107,14 +107,16 @@ func (a *Agent) startTask(ctx context.Context, tc *taskContext, complete chan<-
 		tc.logger.Task().Info("task canceled")
 		return
 	}
-	newDir, err := a.createTaskDirectory(tc)
-	if err != nil {
-		tc.logger.Execution().Errorf("error creating task directory: %s", err)
-		complete <- evergreen.TaskFailed
-		return
+	if tc.runGroupSetup {
+		tc.taskDirectory, err = a.createTaskDirectory(tc)
+		if err != nil {
+			tc.logger.Execution().Errorf("error creating task directory: %s", err)
+			complete <- evergreen.TaskFailed
+			return
+		}
```

Comment:
we talked in person to see if we could merge this, somehow, with the "taskPreCommand" but if that isn't obvious, it's not *that* weird, I think